File: Leetcode_169.java


class Solution {
    public int majorityElement(int[] nums) {
        HashMap<Integer, Integer> freq = new HashMap<>();
        int n = nums.length / 2;

        for (int num : nums) {
            freq.put(num, freq.getOrDefault(num, 0) + 1);
            if (freq.get(num) > n) return num;
        }

        return -1; // Just in case, though problem guarantees a majority
    }
}

File: Leetcode_169.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        freq = {}
        majority_count = len(nums) // 2

        for num in nums:
            if num not in freq:
                freq[num] = 0
            freq[num] += 1

            if freq[num] > majority_count:
                return num
